extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/DevClusterHelper.java

```diff
@@ -153,6 +157,23 @@ public static List<DecoratorBuildItem> createDecorators(String clusterKind,
                         ports,
                         config.ports));
 
+        for (Map.Entry<String, String> annotation : config.getIngress().getAnnotations().entrySet()) {
+            result.add(new DecoratorBuildItem(clusterKind,
+                    new AddAnnotationDecorator(name, annotation.getKey(), annotation.getValue(), INGRESS)));
+        }
+
+        for (IngressRuleConfig rule : config.ingress.rules.values()) {
+            result.add(new DecoratorBuildItem(clusterKind, new AddIngressRuleDecorator(name, port,
```

Comment:
This is also doing more than just dealing with the annotations since it also adds ingress rules (which is fine, by the way, at least in my opinion)